src/api/v4/layer/layer.js

```diff
@@ -63,26 +81,33 @@ var metadataParser = require('./metadata/parser');
  * @memberof carto.layer
  * @api
  */
-function Layer (source, style, options) {
+function Layer (source, style, options = {}) {
   Base.apply(this, arguments);
 
-  options = options || {};
-
   _checkSource(source);
   _checkStyle(style);
 
+  options = _.defaults(options, {
```

Comment:
Why did you change the default assignment to `_.defaults`? 🤔

Reply:
I know that the `visible` check _might not be_ as easy as the other ones, but you can take the chance to make a simple type check, so that you only allow boolean values.

Something like that:
```js
this._visible = _.isBoolean(options.visible) ? options.visible : true;
```

Reply:
It was easier to handle the default value of `visible` being it a boolean, since I can't do `falsy || true` and I didn't want to add a bunch of conditionals, what do you recommend?

Reply:
I recommend what I posted on top of your comment 😂

Reply:
you were faster pressing enter 😂